src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs

```diff
@@ -31,6 +31,43 @@ public static IServiceCollection AddOptions(this IServiceCollection services)
             return services;
         }
 
+        /// <summary>
+        /// Adds services required for using options and enforces options validation check on start rather than in runtime.
+        /// The <seealso cref="OptionsBuilderExtensions.ValidateOnStart{TOptions}(OptionsBuilder{TOptions})"/> extension is called by this method.
+        /// </summary>
+        /// <typeparam name="TOptions">The options type to be configured.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="name">The name of the options instance.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static OptionsBuilder<TOptions> AddOptionsWithValidateOnStart<
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TOptions>(
+            this IServiceCollection services,
+            string? name = null)
+            where TOptions : class
+        {
+            return new OptionsBuilder<TOptions>(services, name ?? Options.Options.DefaultName).ValidateOnStart();
```

Comment:
Shouldn't this call `AddOptions`? I think neither the `OptionsBuilder` constructor nor `ValidateOnStart` do this.

```suggestion
            services.AddOptions();
            return new OptionsBuilder<TOptions>(services, name ?? Options.Options.DefaultName).ValidateOnStart();
```

Reply:
`ValidateOnStart()` does call `optionsBuilder.Services.AddOptions<StartupValidatorOptions>()`